tests/jballerina-unit-test/src/test/resources/test-src/closures/closure.bal

```diff
@@ -751,3 +751,10 @@ function funcAcceptingIntArgs(int... scores) returns int {
     };
     return fn();
 }
+
+function errorConstructorReferenceTest() returns error? {
```

Comment:
We need to assert something, right?

Let's have this function return `function() returns error?` and return `process` here.

And then call this function in another function, get the returned function, call that function and assert that the captured variables are the expected values.